File: atcoder/abc146B_rot_n.ml
(* Vicfred
 * https://atcoder.jp/contests/abc146/tasks/abc146_b
 * string manipulation
 * *)
let n = read_int()
let _ = String.map (fun c -> 
    Char.(chr @@ (code c - code 'A' + n) mod 26 + code 'A')) @@
    read_line() |> print_endline

File: atcoder/abc146B_rot_n.hs
-- uninhm
-- https://atcoder.jp/contests/abc146/tasks/abc146_b
-- implementation

import Data.Char

solve n c =
  chr $ ( (ord c) + n - 65 ) `mod` 26 + 65

main = do
  li <- getLine
  let n = read li
  s <- getLine
  putStrLn $ map (solve n) s

